charts/templates/configmap.yaml

```diff
@@ -10,3 +10,140 @@ data:
   MARKLOGIC_FQDN_SUFFIX: {{ include "marklogic.headlessURL" . }}
   MARKLOGIC_INIT: "true"
   MARKLOGIC_JOIN_CLUSTER: "true"
+---
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: {{ include "marklogic.fullname" . }}-fb-config-map
+  namespace: {{ .Release.Namespace }}
+data:
+  fluent-bit.conf: |
+      [SERVICE]
+        Flush 5
+        Log_Level info
+        Daemon off
+        Parsers_File parsers.conf
+
+      @INCLUDE inputs.conf
+      @INCLUDE filters.conf
+      @INCLUDE outputs.conf
+
+  inputs.conf: |
+      {{- if .Values.logging.trackFiles.errorLogs }}
+      [INPUT]
+        Name tail
+        Path {{ .Values.persistence.mountPath }}/Logs/*ErrorLog.txt
+        Read_from_head true
+        Tag kube.marklogic.logs.error
+        Path_Key path
+        Parser error_parser
+        Mem_Buf_Limit 4MB
+      {{- end }}
+      
+      {{- if .Values.logging.trackFiles.accessLogs }}
+      [INPUT]
+        Name tail
+        Path {{ .Values.persistence.mountPath }}/Logs/*AccessLog.txt
+        Read_from_head true
+        tag kube.marklogic.logs.access
+        Path_Key path
+        Parser access_parser
+        Mem_Buf_Limit 4MB
+      {{- end }}
+
+      {{- if .Values.logging.trackFiles.requestLogs }}
+      [INPUT]
+        Name tail
+        Path {{ .Values.persistence.mountPath }}/Logs/*RequestLog.txt
+        Read_from_head true
+        tag kube.marklogic.logs.request
+        Path_Key path
+        Parser json_parser
+        Mem_Buf_Limit 4MB
+      {{- end }}
+
+      {{- if .Values.logging.trackFiles.crashLogs }}
+      [INPUT]
+        Name tail
+        Path {{ .Values.persistence.mountPath }}/Logs/CrashLog.txt
+        Read_from_head true
+        tag kube.marklogic.logs.crash
+        Path_Key path
+        Mem_Buf_Limit 4MB
+      {{- end }}
+
+      {{- if .Values.logging.trackFiles.auditLogs }}
+      [INPUT]
+        Name tail
+        Path {{ .Values.persistence.mountPath }}/Logs/AuditLog.txt
+        Read_from_head true
+        tag kube.marklogic.logs.audit
+        Path_Key path
+        Mem_Buf_Limit 4MB
+      {{- end }}
+
+  outputs.conf: |
```

Comment:
The options are different for each type of output so there is nothing concrete I could put in the values file without choosing a default type for them. If you wanted that functionality I'd have to make an output for every output that fluent bit supports then have a toggle option for all of them and continue to update the outputs/list of outputs each time fluent bit adds support for a new output if we want to support it. 

There is an example of what to add to the values file for an ElasticSearch output in the values file as well as example of what that should look like in the outputs to give information on where and what to edit for setting up output(s) to try and avoid any confusion given it's an unusual situation.

Reply:
@pengzhouml @rwinieski do you have any ideas?

Reply:
Changing manually files into template folder should be avoided as much as possible.

We can create the configmap.yaml content dynamically from the values.yaml.
As the configmap is only a reference to build fluent-bit.conf and custom_parsers.conf files.

The user could then freely build his own config file.

Reply:
I agree with @jkerr5 and @rwinieski that user should be able to modify all the 4 fluent-bit configurations without needing to change the source code directly. I think 2 ways to do it with values file.

One is to provide the configMapName(or secretName if it includes credential to connect to logging backend in output section) in the values file, then the users are responsible to create the configuration by themselves. We can provide the sample configuration in the document.

The second way is to have a new ConfigMap and put all the configuration in the values file directly. for example, the new config file could be something like:
```
data: 
  fluent-bit.conf: |
    {{- .Values.logCollection.config | nindent 4 }}
```
the values file could be something like:
```
logCollection:
  config: |
    [Service]
      .....
    [INPUT]
      .....
```
We can choose to put all configs in one file or separate the configuration in 4 files.

Then we also need to provide the documentation on how to set the fluent bit config in the values file.

Reply:
@jkerr5 @pengzhouml @rwinieski Fluent bit's output configuration is now set inside of the values file witch should solve the issue James initially brought up. It is my opinion that the outputs section is the only one that customers should be encouraged to change and therefor should be the only one specified in the values file but if there is good reason to move the configurations of some or all of the other fluent bit config sections to the values file as-well that can be done.

Reply:
Just had a talk with Ilan and we will move all the configuration to the values file. It did make the values file pretty big, however, it provides the most flexibility if user would like to change any of the configuration later on. 

@jkerr5 and @rwinieski please let us know if you have and question or concern regarding this decision.

Reply:
Update, moving the inputs and filters sections to the values file has been problematic. This is because the values file can't reference itself so we lose the ability to use the if statements for toggling witch files to collect (not a big issue because now the customer can just change the inputs section, but more importantly we lose access to the mountPath variable, the release namespace, as well as the pod name variable. So to move those two sections we'd have to lose some functionality or thing of a different way of giving the customer the configurations other than directly in the values file.

Reply:
The output section is the main one I was concerned about. I think the higher-level abstraction of having the different on/off switches for the different log files is nice. Being able to control which files at the app server level is a good follow-up improvement but let's create a separate story for that.